File: 0253-meeting-rooms-ii/0253-meeting-rooms-ii.cpp
class Solution {
public:
    int minMeetingRooms(vector<vector<int>>& intervals) {
        if(intervals.size()==0){
            return 0;
        }
        sort(intervals.begin(),intervals.end());
        priority_queue<int,vector<int>,greater<int>>pq;
        int ans=1;
        for(auto it:intervals){
            int end= it[1];
            while(!pq.empty() && it[0]>=pq.top()){
                pq.pop();
            }
            pq.push(end);
            if(pq.size() >ans){
                ans=pq.size();
            }
        }
        return ans;
    }
};

File: 0253-meeting-rooms-ii/0253-meeting-rooms-ii.java
class Solution {

    public int minMeetingRooms(int[][] intervals) {

        PriorityQueue<Integer> pq = new PriorityQueue<>();

        Arrays.sort(intervals, (a,b) -> a[0]-b[0]);
        pq.add(intervals[0][1]);
        int maxCount = 1;
        for(int i = 1; i<intervals.length; i++){
            while(!pq.isEmpty() &&pq.peek()<=intervals[i][0]){
                pq.poll();
            }
            pq.add(intervals[i][1]);

            maxCount = Math.max(maxCount, pq.size());
        }
        return maxCount;
    }
}